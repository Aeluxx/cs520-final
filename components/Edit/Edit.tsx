import { Typography } from '@mui/material'
import { Box } from '@mui/system'
import { Editor } from '@tinymce/tinymce-react'
import { useRef, useState, useEffect } from 'react'
import { useRouter } from 'next/router'
import useDebounce from '../../hooks/useDebounce'
import axios from 'axios'
import useAxios from 'axios-hooks'

export default function Edit() {
  const router = useRouter()
  const noteId = router.query.id
  const editorRef = useRef<any>(null)
  const [content, setContent] = useState('')
  const debouncedContent = useDebounce(content, 1000)
  const [{ data, loading }] = useAxios({ url: 'http://localhost:3000/api/note', method: 'GET', params: { id: noteId } })
  console.log('data', data)
  useEffect(() => {
    if(loading) return
    setContent(data?.content)
  }, [data])

  const updateNote = async () => {
    const x = await axios.patch('http://localhost:3000/api/note', { id: noteId, content: debouncedContent })
    console.log('x', x)
  }

  // useEffect(() => {
  //   updateNote()
  // }, [debouncedContent])

  return (
    <Box
      sx={{
        m: 3,
      }}>
      <p>{debouncedContent}</p>
      <Editor
        apiKey={process.env.NEXT_PUBLIC_TINY_KEY}
        onInit={(evt, editor) => (editorRef.current = editor)}
        value={content}
        onEditorChange={val => setContent(val)}
        init={{
          height: 500,
          menubar: false,
          plugins: [
            'advlist autolink lists link image charmap print preview anchor',
            'searchreplace visualblocks code fullscreen',
            'insertdatetime media table paste code help wordcount',
          ],
          toolbar:
            'undo redo | formatselect | ' +
            'bold italic backcolor | alignleft aligncenter ' +
            'alignright alignjustify | bullist numlist outdent indent | ' +
            'removeformat | help',
          content_style: 'body { font-family:Helvetica,Arial,sans-serif; font-size:14px }',
        }}
      />
    </Box>
  )
}
