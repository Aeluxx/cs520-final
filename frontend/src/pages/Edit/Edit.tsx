import React from 'react'
import { useParams } from 'react-router-dom'

export default function Edit() {
  const { docId } = useParams()
  return (
    <div>Editing document {docId}...</div>
  )
}
