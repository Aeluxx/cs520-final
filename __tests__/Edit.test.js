import React from 'react';
import '@testing-library/jest-dom';
import Edit from '../components/Edit/Edit';
import ShallowRenderer from 'react-test-renderer/shallow';

let realUseContext;
let useContextMock;

beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
});

afterEach(() => {
    React.useContext = realUseContext;
});

it("Edit renders without error", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const edit = shallow.render(<Edit />);
});

it("Edit matches Snapshot", () => {
    useContextMock.mockReturnValue("Test Value");
    const shallow = new ShallowRenderer();
    const edit = shallow.render(<Edit />);
    const result = shallow.getRenderOutput();
    expect(result).toMatchSnapshot();
});