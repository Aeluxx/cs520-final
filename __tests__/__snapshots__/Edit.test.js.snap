// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit matches Snapshot 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "m": 3,
    }
  }
>
  <ForwardRef(TextField)
    onChange={[Function]}
    size="small"
    sx={
      Object {
        "display": "block",
        "my": 1,
      }
    }
    value={null}
  />
  <ForwardRef(Typography)
    variant="caption"
  >
    Last updated 
    a few seconds ago
  </ForwardRef(Typography)>
  <Editor
    cloudChannel="6"
    init={
      Object {
        "content_style": "body { font-family:Helvetica,Arial,sans-serif; font-size:14px }",
        "height": 500,
        "menubar": false,
        "plugins": Array [
          "advlist autolink lists link image charmap print preview anchor",
          "searchreplace visualblocks code fullscreen",
          "insertdatetime media table paste code help wordcount",
        ],
        "toolbar": "undo redo | formatselect | bold italic backcolor | alignleft aligncenter alignright alignjustify | bullist numlist outdent indent | removeformat | help",
      }
    }
    onEditorChange={[Function]}
    onInit={[Function]}
    value=""
  />
</ForwardRef(Box)>
`;
